state.go

```diff
@@ -668,6 +799,12 @@ func (s *State) MessageAdd(message *Message) error {
 	if len(c.Messages) > s.MaxMessageCount {
 		c.Messages = c.Messages[len(c.Messages)-s.MaxMessageCount:]
 	}
+
+	if c.IsThread() {
+		c.MessageCount++
+		c.LastMessageID = message.ID
+	}
```

Comment:
We don't do this for normal channels, and the eventual consistency model of Discord's events means that these might not even be true. I'd just leave these properties alone and leave it to the semantics defined in the API.

Reply:
I'm not sure if this is a good idea or not, because this might be confusing why the Thread in the state does not reflect on the last message. I mean, yeah, you could fetch the thread, but still, that is confusing.

Reply:
Alright, I removed this, but not sure if this is a good idea. It might be confusing for a user why the thread does not contain the last message, like normal channels.

Reply:
Also, since we're removing this we might need to remove this logic from `MessageRemove` as well

Reply:
This is marked as "outdated" so I'm unclear whether it's resolved or not

Reply:
It's not quite resolved, since I'm personally unsure about removal of this.

Reply:
I don't think my opinion has changed here. I don't think that we should modify data from Discord that makes it impossible for a user to see what has canonically been sent. Thread updates should correct this message count. (I would agree that taking the logic out of message removal would make sense, as well.)